__tests__/app.test.js

```diff
@@ -45,25 +37,133 @@ describe('/api', () => {
+    test('GET: 200 returns all comments associated with a specific article_id in ascending order based on creation date', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then((response) => {
+            const {comments} = response.body
+            comments.forEach(comment => {
+                expect(comment).toMatchObject({
+                    comment_id : expect.any(Number),
+                    body: expect.any(String),
+                    article_id: expect.any(Number),
+                    author: expect.any(String),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String)
+                    })
                 })
+                expect(comments).toBeSorted({ descending: false });
             })
+    });
+    test('POST:201', () => {
+        const newComment = {
+            author: 'butter_bridge',
+            comment: 'did  you know that peppa pig is actually vegetarian?',
+            article_id: 1
+            }
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            expect(response.body.comment.author).toBe(newComment.author);
+        expect(response.body.comment.body).toBe(newComment.comment);
+        expect(response.body.comment.article_id).toBe(newComment.article_id);
+        })
+    });
+})
+
+
+describe('Error Handling', () => {
+    describe('404: Bad Path', () => {
+        test('GET:404 sends an appropriate status and error message when given an invalid url', () => {
+        return request(app)
+        .get('/api/articules')
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe('Invalid URL');
+            })
+        });
+        test('GET:404 sends an appropriate status and error message when given an invalid url', () => {
+            return request(app)
+            .get('/api/topices')
+            .expect(404)
+            .then((response) => {
+            expect(response.body.msg).toBe('Invalid URL');
+            });
+        });
+        test('GET:404 sends an appropriate status and error message when provided with a non-existent but valid id', () => {
+            return request(app)
+            .get('/api/articles/999')
+            .expect(404)
+            .then((response) => {
+            expect(response.body.msg).toBe('Article does not exist');
+            });
+        });
+    });
+    test('POST: 404 Username not found', () => {
+        const newComment = {
+            author: 'peppapig',
+            comment: 'winner winner chicken dinner',
+            article_id: 2
+            }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(newComment)
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe('Username not found');
```

Comment:
We also want to add testing for the post end point for...
Status 201, ignores unnecessary properties
Status 400, invalid ID, e.g. string of "not-an-id"
Status 404, non existent ID, e.g. 0 or 9999
Status 400, missing required field(s), e.g. no username or body properties